.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,24 @@
+<!--
+Adding a new resource or datasource? Use this checklist to get started: https://github.com/hashicorp/terraform-provider-hcp/blob/main/contributing/checklist-resource.md
```

Comment:
Worried this comment might get ignored, so I'd move the link into the checklist I described below.

Reply:
Since we're omitting a `Definition of Done` for now, I'm going to leave this here in the comment, but agreed that we should be ready to tweak this if we are noticing contributors skipping past it